src/Middleware/RequestBodyBufferMiddleware.php

```diff
@@ -30,27 +30,28 @@ public function __construct($sizeLimit = null)
 
     public function __invoke(ServerRequestInterface $request, $stack)
     {
+        $sizeLimit = $this->sizeLimit;
         $body = $request->getBody();
 
         // request body of known size exceeding limit
         if ($body->getSize() > $this->sizeLimit) {
-            return new Response(413, array('Content-Type' => 'text/plain'), 'Request body exceeds allowed limit');
+            $sizeLimit = 0;
         }
 
         if (!$body instanceof ReadableStreamInterface) {
             return $stack($request);
         }
 
-        return Stream\buffer($body, $this->sizeLimit)->then(function ($buffer) use ($request, $stack) {
+        return Stream\buffer($body, $sizeLimit)->then(function ($buffer) use ($request, $stack) {
             $stream = new BufferStream(strlen($buffer));
             $stream->write($buffer);
             $request = $request->withBody($stream);
 
             return $stack($request);
-        }, function($error) {
+        }, function ($error) use ($stack, $request) {
             // request body of unknown size exceeding limit during buffering
             if ($error instanceof OverflowException) {
-                return new Response(413, array('Content-Type' => 'text/plain'), 'Request body exceeds allowed limit');
+                return $stack($request);
```

Comment:
Awesome, I think these changes make perfect sense :+1: 

One question that remains is, when should the next handler be called? It's my understanding that this middleware handler should still wait for the complete request to `end` before proceeding. This basically means that this middleware will wait for the complete request body anyway and that we simply ignore its contents.

Also, I wonder what should happen with `Content-Length` (remove?) and `getBody()->getSize()` (set to `0`?) etc. We should probably ensure to only pass updated values to the next consumer.

Can you update this to include some tests and documentation for this? :+1:

Reply:
> One question that remains is, when should the next handler be called? It's my understanding that this middleware handler should still wait for the complete request to end before proceeding. This basically means that this middleware will wait for the complete request body anyway and that we simply ignore its contents.

Adding tests next so we can ensure the right behavior.

> Can you update this to include some tests and documentation for this? :+1:

It's on the list, but wanted code out first so we can discuss before spending time on documentation.